core/src/main/groovy/noe/tomcat/configure/ConnectorConfiguratorTomcat.groovy

```diff
@@ -124,29 +124,33 @@ class ConnectorConfiguratorTomcat {
   }
 
   private void updateExistingConnector(Node connector, Node newConnector) {
-    // update attributes
+    // update connector attributes
     newConnector.attributes().each { attribute ->
       connector.@"${attribute.key}" = attribute.value
     }
 
-    // update nodes
     newConnector.each { Node newSubelement ->
-      String UpgradeProtocol = ConnectorUpgradeProtocolTomcat.ELEMENT_NAME
-
-      if (newSubelement.name() == UpgradeProtocol) {
-        if (connector.find { it.name() == UpgradeProtocol } == null) {
-          // create new element
-          connector.appendNode(newSubelement, newSubelement.attributes(), newSubelement.value())
-        } else {
-          // upgrade existing element
-          connector.findAll { it.name() == UpgradeProtocol }.each { upgradeProtocol ->
-            newSubelement.attributes() { attribute ->
-              upgradeProtocol.@"${attribute.key}" = attribute.value
-            }
-          }
-        }
+      updateInnerElements(connector, newSubelement)
+    }
+  }
+
+  /**
+   *
+   * @param connector
+   * @param newSubElement
+   *
+   * Search for existing inner element to remove and replace with new one
+   * Note: SSLHostConfig inner elements can exist more than one
```

Comment:
What implication of this? What anyone should know?

Reply:
What do you mean? I used it as an explanation comment for future development.

Reply:
Ok, I see.